packages/web-components/src/button/button.template.ts

```diff
@@ -11,7 +11,7 @@ import { ButtonOptions } from './button.options.js';
 export function buttonTemplate<T extends Button>(options: ButtonOptions = {}): ElementViewTemplate<T> {
   return html<T>`
     <template
-      tabindex="${x => (x.disabled ? -1 : x.tabIndex ?? 0)}"
+      tabindex="${x => (x.disabled ? -1 : x.tabIndex)}"
```

Comment:
Instead of setting the `tabindex` attribute here, which will then clobber the `tabIndex` property, we can set it to `null` to remove the attribute when it's disabled:
 
```suggestion
      tabindex="${x => (x.disabled ? null : x.tabIndex ?? 0)}"
```

There may need to be some additional checks for `disabledFocusable`.